### PR TITLE
feat: add APIs of ExecuteOfflineQuery, ImportOnlineData and ImportOfflineData for CLI

### LINKED ISSUE #409: Support passing Spark parameters for TaskManager
Currently we can submit batch jobs by passing the SQL text to run in yarn cluster. The other spark parameters are ignored since TaskManager API does not support dynamic configuration. We should support passing dynamic Spark parameters for Spark jobs.

### PR DESCRIPTION
**Pull request by tobegit3hub**
* Read execute mode and run ExecuteOfflineQuery
* Read execute mode and run ImportOnlineData or ImportOfflineData
* Fix the path of input file for Spark in openmldb-batch unit tests
* PrintOfflineTableInfo when desc the table in CLI
* Pass the default_db from CLI session to TaskManager APIs
* Resolve https://github.com/4paradigm/OpenMLDB/issues/409

### CODE REVIEW

File: src/sdk/sql_cluster_router.h [hunk 1]
```diff
@@ -209,6 +209,21 @@ class SQLClusterRouter : public SQLRouter {
 
     ::openmldb::base::Status StopJob(const int id, ::openmldb::taskmanager::JobInfo& job_info) override;
 
+    ::openmldb::base::Status ExecuteOfflineQuery(const std::string sql,
```
Comment (dl239) [inline src/sdk/sql_cluster_router.h]: const &

**Review by dl239 — APPROVED** [review]

File: src/client/taskmanager_client.cc [hunk 2]
```diff
@@ -81,5 +81,75 @@ ::openmldb::base::Status TaskManagerClient::StopJob(const int id, ::openmldb::ta
     }
 }
 
+::openmldb::base::Status TaskManagerClient::RunBatchAndShow(const std::string sql,
```
Comment (vagetablechicken) [inline src/client/taskmanager_client.cc]: better to use `const std::string&`, `const std:map<>&`

Comment (tobegit3hub) [reply src/client/taskmanager_client.cc]: Yes and fixed.

File: src/client/taskmanager_client.cc [hunk 3]
```diff
@@ -81,5 +81,75 @@ ::openmldb::base::Status TaskManagerClient::StopJob(const int id, ::openmldb::ta
...
+    bool ok = client_.SendRequest(&::openmldb::taskmanager::TaskManagerServer_Stub::ImportOfflineData, &request,
+                                  &response, FLAGS_request_timeout_ms, 1);
+
+    if (ok && response.code() == 0) {
+        if (response.has_job()) {
+            job_info.CopyFrom(response.job());
+        }
+        return ::openmldb::base::Status(response.code(), response.msg());
+    } else {
+        return ::openmldb::base::Status(-1, "Fail to request TaskManager server");
```
Comment (vagetablechicken) [inline src/client/taskmanager_client.cc:150]: use `ReturnCode`? e.g. {kError, "Fail to request TaskManager server"}

File: src/cmd/sql_cmd.h [hunk 4]
```diff
@@ -79,6 +79,25 @@ ::openmldb::sdk::SQLClusterRouter* sr = nullptr;
 using VariableMap = std::map<std::string, std::string>;
 VariableMap session_variables = {VariableMap::value_type("execute_mode", "online")};
 
+bool isOnlineMode() {
```
Comment (vagetablechicken) [inline src/cmd/sql_cmd.h]: Uppercase -> `IsOnlineMode`

File: src/cmd/sql_cmd.h [hunk 5]
```diff
@@ -79,6 +79,25 @@ ::openmldb::sdk::SQLClusterRouter* sr = nullptr;
...
+    for (auto& pair : session_variables) {
```
Comment (vagetablechicken) [inline src/cmd/sql_cmd.h]: session_variables must have the key `execute_mode`, so you can use `session_variables["execute_mode"]`, or more robustly use session_variables.find("execute_mode"), no need to loop

File: src/cmd/sql_cmd.h [hunk 6]
```diff
@@ -1275,10 +1348,33 @@ void HandleSQL(const std::string& sql) {
         }
         case hybridse::node::kPlanTypeLoadData: {
             auto plan = dynamic_cast<hybridse::node::LoadDataPlanNode*>(node);
-            std::string error;
-            if (!HandleLoadDataInfile(plan->Db(), plan->Table(), plan->File(), plan->Options(), &error)) {
-                std::cout << "ERROR: Load data failed. " << error << std::endl;
-                return;
+
+            if (FLAGS_host.empty() || FLAGS_port == 0) {
```
Comment (vagetablechicken) [inline src/cmd/sql_cmd.h]: use `cs->IsClusterMode()`?

File: src/cmd/sql_cmd.h [hunk 7]
```diff
@@ -1244,12 +1303,26 @@ void HandleSQL(const std::string& sql) {
...
+                }
             } else {
-                PrintResultSet(std::cout, rs.get());
+                // Run offline query
+                ::openmldb::taskmanager::JobInfo job_info;
+                std::map<std::string, std::string> config;
+                auto status = sr->ExecuteOfflineQuery(sql, config, db, job_info);
+
+                std::vector<::openmldb::taskmanager::JobInfo> job_infos;
+                if (status.OK() && job_info.id() > 0) {
```
Comment (vagetablechicken) [inline src/cmd/sql_cmd.h:1315]: If status is not ok or job info invalid, should we print the error?

File: src/client/taskmanager_client.cc (hunk 3 above)
Comment (tobegit3hub) [reply src/client/taskmanager_client.cc:150]: We will refactor to use the error code later.

File: src/cmd/sql_cmd.h (hunk 4 above)
Comment (tobegit3hub) [reply src/cmd/sql_cmd.h]: Okay.

File: src/cmd/sql_cmd.h (hunk 5 above)
Comment (tobegit3hub) [reply src/cmd/sql_cmd.h]: Okay.

File: src/cmd/sql_cmd.h (hunk 6 above)
Comment (tobegit3hub) [reply src/cmd/sql_cmd.h]: Okey.

File: src/cmd/sql_cmd.h (hunk 7 above)
Comment (tobegit3hub) [reply src/cmd/sql_cmd.h:1315]: We will redefine the output message for not-ok processes laster.

**Review by vagetablechicken — APPROVED** [review]

**Comment by tobegit3hub** [comment]
[CLI for offline jobs with unified metadata store](https://trello.com/c/rcSLPVcy/21-cli-for-offline-jobs-with-unified-metadata-store)

**Pull request merged**